### DISCUSSION TITLE
Seperate use of Rx and Tx lines when using micropyGPS

### DISCUSSION THREAD

**Discussion by Vesta6839** [micropython/micropython#12874] [RP2040 / Pico] [answered]
My first attempt to program a Raspberry Pi Pico is using MicroPython to read and parse a NMEA $GPRMC sentence from a GPS using micropyGPS. This uses only the Rx line of the chosen UART, will I be able to use the Tx line of that UART to send data to another device ? Or will micropyGPS ´claim´ this Tx line for it´s own use ?

**Comment by robert-hh** [comment, marked as answer]
Hardly. And even if, RX and TX of a UART have the same settings of e.g. baud rate, bits, .... At the Pico you can as well use the PIO for UART data transfer. If you enable PIO transfer after setting up the UART, it should be possible to override the use of the TX pin by the UART.
For the PIO UART example. see https://github.com/raspberrypi/pico-micropython-examples/tree/master/pio
